### ISSUE TITLE
Add support for the Gateway object

### ISSUE THREAD

**Issue by baruchoxman** [andrewvillazon/pbipy#45]
The functionality I'm looking for is to iterate over gateways and their data sources:
https://learn.microsoft.com/en-us/rest/api/power-bi/gateways/get-gateways
https://learn.microsoft.com/en-us/rest/api/power-bi/gateways/get-datasources

Thank you and keep doing the great job with this package!

**Comment by andrewvillazon** [comment]
@baruchoxman - Have added Gateways. Feel free to take a look and let me know if there's any issues. To install:

```
pip install git+https://github.com/andrewvillazon/pbipy@45-add-support-for-the-gateway-object
```

To retrieve the Gateways there is a couple of new methods on the pbi client object: `PowerBI.gateway` for a single gateway, and `PowerBI.gateways` for all Gateways. Both return `Gateway` objects which have a `Gateway.datasources` method for all attached datasources.

**Comment by baruchoxman** [comment]
Thanks, that works, feel free to merge!

**Comment by andrewvillazon** [comment]
Merged and released in: [v2.11.0](https://github.com/andrewvillazon/pbipy/releases/tag/v2.11.0)
